addon/components/feature-controls.js

```diff
@@ -1,42 +1,59 @@
-import Component from '@ember/component'
+import Component from '@glimmer/component'
+import { get, set, action } from '@ember/object'
 import { inject as service } from '@ember/service'
-import { action, set } from '@ember/object'
+import { camelize } from '@ember/string'
 import { assign } from '@ember/polyfills'
-import config from 'ember-get-config'
 import windowUtil from 'ember-feature-controls/utils/window'
+import { getOwner } from '@ember/application'
 
-const { featureFlags, featureControls } = config
+export default class FeatureControlsComponent extends Component {
+  @service features
+  @service featureControlsStorage
 
-export default Component.extend({
-  tagName: '',
-  features: service(),
-  featureControlStorage: service(),
-  showRefresh: true,
-  showReset: true,
-  featureControls,
-  featureFlags,
+  get featureFlags() {
+    return this.args.featureFlags ? this.args.featureFlags : this._featureFlags
+  }
 
-  init() {
-    this._super(...arguments)
+  get featureControls() {
+    return this.args.featureControls
+      ? this.args.featureControls
+      : this._featureControls
+  }
+
+  get showRefresh() {
+    return this.args.showRefresh ?? true
+  }
+
+  get showReset() {
+    return this.args.showReset ?? true
+  }
+
+  constructor(...args) {
+    super(...args)
```

Comment:
Any reason to not use ?
```suggestion
  constructor() {
    super(...arguments)
```

Reply:
A bad habit